ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,306 @@
-##########################################################################
-sub Configure_Config {
-##########################################################################
-  my ($self,$config)=@_;
-
-# bla. Only set interesting stuff right now -> OPTIONS, -nosettime, verbose, enabled
-
-  my $node;
-  my $options = '';
-  my $have_config = 0;
-  my $ret =1 ;
-  if($config->elementExists("$mypath/options")) {
-    $node = $config->getElement("$mypath/options");
-    while ($node->hasNextElement()) {
-      my $next = $node->getNextElement();
-      my $nextname = $next->getName();
-      my $nextvalue = $next->getValue();
-      $options .= " " if $options;
-      $options .= "-$nextname $nextvalue";
-      $have_config = 1;
-    }
-  }
-  # "settime" option
-  if($config->elementExists("$mypath/settime")) {
-    $node = $config->getElement("$mypath/settime");
-    $have_config = 1;
-    my $settime = $node->getValue();
-    if(! $settime) {  # caution, not set = assume NTP
-      $options .= " " if $options;
-      $options .= "-nosettime"
-    }
-  }
-  if($have_config) {
-    $ret = NCM::Check::lines($sysconfig_afs,
-			     linere => '^#.*$compname-options|OPTIONS=.*',
-			     goodre => '^#.*$compname-options|OPTIONS=\s*"'.quotemeta($options).'"',
-			     good   => "# changed by $compname-options\nOPTIONS=\"".$options."\"",
-			     keep   => 'first',
-			     add    => 'last',
-			     backup => '.old_options');
-  } else {
-    $self->info("no OPTIONS configuration for $sysconfig_afs, skipping");
-  }
-
-  # cache size is configured in a separate function
-
-  if($config->elementExists("$mypath/verbose")) {
-    $node = $config->getElement("$mypath/verbose");
-    my $verbose = $node->getValue() ? '-verbose' : '';
-    $ret &= NCM::Check::lines($sysconfig_afs,
-			     linere => '^#.*$compname-verbose|VERBOSE=.*',
-			     goodre => '^#.*$compname-verbose|VERBOSE=\s*'.quotemeta($verbose),
-			     good   => "# changed by $compname-verbose\nVERBOSE=".$verbose,
-			     keep   => 'first',
-			     add    => 'last',
-			     backup => '.old_verbose');
-  }
-
-  if($config->elementExists("$mypath/enabled")) {
-    $node = $config->getElement("$mypath/enabled");
-    my $enabled = $node->getValue(); #syntax needs to be on/off, not yes/no
-    if ($enabled =~ /^off$|^no$/i) {
-      $enabled = 'off';
-    } else {
-      $enabled = 'on'; #default
+    my $proc = CAF::Process->new( [ "fs", "getcacheparms" ], log => $self, keeps_state => 1 );
+    my $output = $proc->output();
+    if ( !( $? >> 8 ) && $output =~ /AFS using \d+ of the cache's available (\d+) (\w+) byte blocks/ ) {
+        if ( $2 ne "1K" ) {
+            $self->error("Cannot handle $2 (non-1K) AFS cache block sizes");
+            return 0;
+        }
+        $run_cache = $1;
+    }
+    else {
+        $self->warn("Cannot determine current AFS cache size, changing only config file");
+    }
+
+    my $afs_cacheinfo_fh = CAF::FileEditor->new( $AFS_CACHEINFO, log => $self );
```

Comment:
ok, can you add the  following:
1. an item with future milestaone that says to use FileReader in afsctl
2. add a `# TODO use FileReader`
3. do a `$fs->cancel()` after each `->new` (it's basically what filereader does)